### PR TITLE
Server > Toggling Feature Availability

### PR DESCRIPTION
**Pull request by mjhilton**
Documents available options for toggling Feature availability, the intent of each, and how you would choose the right option for your situation.

Spawned from [🔒 Slack Discussion](https://octopusdeploy.slack.com/archives/CDANN5QLT/p1637727878212900) (internal)

### CODE REVIEW

File: OctopusServer/TogglingFeatures.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Sometimes we don't want a feature to be available to all customers, immediately upon deployment of the code. We may want to slowly trial an idea with some early alpha/beta-testers, or we might want a kill-switch to be able to react quickly in a way that doesn't require redeployment.
+
+For this, we have a range of different types of feature flags and toggles that control the availability of our features.
+
+## Octopus Server Portal > Configuration > Settings
+Configuration Settings are collections of values that control a set of behaviour. They allow Customers to control and configure things like Authentication mechanism (eg Octopus SSO using AAD, which requires a number of different identifiers, secrets and URLs to get working)
+
+Each of these settings is declared by creating a new class that derives from [ConfigurationSettings<,,>](https://github.com/OctopusDeploy/ServerExtensibility/blob/8f7387ccf824972af3ac6bde8568f379bb8987c7/source/Server.Extensibility/Extensions/Infrastructure/Configuration/ConfigurationSettings.cs).
+
+Configuration Settings are persisted to the `Configuration` table.
```
Comment (slewis74) [inline OctopusServer/TogglingFeatures.md:11]: A minor thing, just for completeness, not everything in that section is in the Configuration table. There is also [ExtensionConfigurationSettings](https://github.com/OctopusDeploy/ServerExtensibility/blob/main/source/Server.Extensibility/Extensions/Infrastructure/Configuration/ExtensionConfigurationSettings.cs), which is a subtype of ConfigurationSettings. They get rendered in the same part of the UI but are stored in the ExtensionConfiguration table. (the Auth providers are all stored in ExtensionConfiguration, as an example).
So they're slightly different mechanically, but same same conceptually. Not sure if it actually warrants calling out in here or not.

File: OctopusServer/TogglingFeatures.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+* You only need a simple enabled/disabled switch
+* You need to toggle something at the infrastructure-level of Server
+* You only need to enable the toggle for a small number of instances
+* You intend the toggle to be short-lived (eg < 2 months)
+
+Do not use these if:
+* Your toggle will be long-lived or permanent
+* You want Customers to be able to see/control feature availability
+* Your toggle requires additional related settings (eg a modifiable endpoint, threshold values etc) to function
+
```
Comment (slewis74) [inline OctopusServer/TogglingFeatures.md:86]: - Your toggle needed a consistent value across all nodes in a HA cluster

File: OctopusServer/TogglingFeatures.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Use these if:
+* You only need a simple enabled/disabled switch
+* You need to toggle something at the infrastructure-level of Server
+* You only need to enable the toggle for a small number of instances
+* You intend the toggle to be short-lived (eg < 2 months)
+
+Do not use these if:
+* Your toggle will be long-lived or permanent
+* You want Customers to be able to see/control feature availability
+* Your toggle requires additional related settings (eg a modifiable endpoint, threshold values etc) to function
```
Comment (slewis74) [inline OctopusServer/TogglingFeatures.md:85]: - Your toggle needed a consistent value across all nodes in a HA cluster

File: OctopusServer/TogglingFeatures.md (hunk 1 above)
Comment (mjhilton) [reply OctopusServer/TogglingFeatures.md]: Done, thanks @slewis74

File: OctopusServer/TogglingFeatures.md (hunk 3 above)
Comment (mjhilton) [reply OctopusServer/TogglingFeatures.md:87]: 👍

**Pull request merged**